opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -55,3 +58,45 @@ def __hash__(self):
 
 
 _EMPTY_RESOURCE = Resource({})
+
+
+class ResourceDetector:
+    def __init__(self, crash_on_error=False):
+        self.crash_on_error = crash_on_error
```

Comment:
What's the use case here?

Reply:
We want the user to have the ability to choose whether or not to ignore errors. i.e. if its vital to scrape resources info or not.